backend/app/tasks/knowledge_base.py

```diff
@@ -55,10 +55,18 @@ def import_documents_from_kb_datasource(kb_id: int, data_source_id: int):
             )
 
             for document in loader.load_documents():
+                # When content of document is too long, truncate it when saving to database.
+                # We can use original content to build index later.
+                original_content = document.truncate_content(max_length=2*1024*1024)
+                if original_content is not None:
+                    logger.info(f"Truncate document #{document.id} from data source #{data_source_id}")
+                    truncated_content_length = document.get_content_length()
+                    logger.info(f"original content length: {len(original_content)}, truncated content length: {truncated_content_length}")
+
                 session.add(document)
                 session.commit()
 
-                build_index_for_document.delay(kb_id, document.id)
+                build_index_for_document.delay(kb_id, document.id, original_content)
```

Comment:
It seems to increase Redis' memory overhead? Can we refer to Git Large File Storage (LFS) for storing references to large files in the database instead of the raw content?

Reply:
We can set a max-upload-size parameters to protect redis and suppose user(admin) have enough sense that he/her SHOULD not set it a very big value(somehow bigger than redis's ram size)

So currenttly I think it looks good to me.